meilisearch-http/src/routes/indexes/documents.rs

```diff
@@ -165,127 +124,76 @@ pub struct UpdateDocumentsQuery {
     primary_key: Option<String>,
 }
 
-pub async fn add_documents_json(
-    meilisearch: GuardedData<Private, MeiliSearch>,
-    path: web::Path<IndexParam>,
-    params: web::Query<UpdateDocumentsQuery>,
-    body: Payload,
-) -> Result<HttpResponse, ResponseError> {
-    document_addition(
-        meilisearch,
-        path,
-        params,
-        body,
-        DocumentAdditionFormat::Json,
-        IndexDocumentsMethod::ReplaceDocuments,
-    )
-    .await
-}
-
-pub async fn add_documents_ndjson(
-    meilisearch: GuardedData<Private, MeiliSearch>,
-    path: web::Path<IndexParam>,
-    params: web::Query<UpdateDocumentsQuery>,
-    body: Payload,
-) -> Result<HttpResponse, ResponseError> {
-    document_addition(
-        meilisearch,
-        path,
-        params,
-        body,
-        DocumentAdditionFormat::Ndjson,
-        IndexDocumentsMethod::ReplaceDocuments,
-    )
-    .await
-}
-
-pub async fn add_documents_csv(
+pub async fn add_documents(
     meilisearch: GuardedData<Private, MeiliSearch>,
     path: web::Path<IndexParam>,
     params: web::Query<UpdateDocumentsQuery>,
     body: Payload,
+    req: HttpRequest,
 ) -> Result<HttpResponse, ResponseError> {
+    debug!("called with params: {:?}", params);
     document_addition(
+        req.headers()
+            .get("Content-type")
+            .map(|s| s.to_str().unwrap_or("unkown")),
         meilisearch,
-        path,
-        params,
+        path.into_inner().index_uid,
+        params.into_inner().primary_key,
         body,
-        DocumentAdditionFormat::Csv,
         IndexDocumentsMethod::ReplaceDocuments,
     )
     .await
 }
 
-pub async fn update_documents_json(
-    meilisearch: GuardedData<Private, MeiliSearch>,
-    path: web::Path<IndexParam>,
-    params: web::Query<UpdateDocumentsQuery>,
-    body: Payload,
-) -> Result<HttpResponse, ResponseError> {
-    document_addition(
-        meilisearch,
-        path,
-        params,
-        body,
-        DocumentAdditionFormat::Json,
-        IndexDocumentsMethod::UpdateDocuments,
-    )
-    .await
-}
-
-pub async fn update_documents_ndjson(
+pub async fn update_documents(
     meilisearch: GuardedData<Private, MeiliSearch>,
     path: web::Path<IndexParam>,
     params: web::Query<UpdateDocumentsQuery>,
     body: Payload,
+    req: HttpRequest,
 ) -> Result<HttpResponse, ResponseError> {
+    debug!("called with params: {:?}", params);
```

Comment:
Do you want to keep that?

Reply:
Yes, this yes, this is part of every route I think. @Irevoire made a PR some time ago to add them

Reply:
Yes, this is in the spec about the logs.